### ISSUE TITLE
passTLSClientCert Middleware should not Escape + and / chars in base64 encoded certificate

### ISSUE THREAD

**Issue by DanTheManSWE** [traefik/traefik#9379]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What did you do?

When passing a certificate from Traefik to backend using passTLSClientCert Middleware it will escape "+" and "/" characters in the base64 encoded certificate, which in turn breaks the certificate. 

The sanitize function in https://github.com/traefik/traefik/blob/master/pkg/middlewares/passtlsclientcert/pass_tls_client_cert.go
returns the certificate using url.QueryEscape which changes "+" to "%2B" and "/" to "%2F".

Instead it should only return the sanitized certificate without escaping. 

### What did you see instead?

The sanitize function in https://github.com/traefik/traefik/blob/master/pkg/middlewares/passtlsclientcert/pass_tls_client_cert.go
returns the certificate using url.QueryEscape which changes "+" to "%2B" and "/" to "%2F".

### What version of Traefik are you using?

version 2.6.0

### What is your environment & configuration?

```yaml
# (paste your configuration here)
```

Add more configuration information here.
Not applicable

**Comment by rtribotte** [comment]
Hello @DanTheManSWE,

Thanks for your interest in Traefik!

This is expected, please see https://doc.traefik.io/traefik/middlewares/http/passtlsclientcert/#pem.

Why do you think Traefik should not escape the certificate?

**Comment by DanTheManSWE** [comment]
The certificate is already encoded in base64 which is ok to send in a header.
If you use an octet percent-encoding on the base64 string it will replace "+" with "%2B" and "/" with "%2F" which make the certificate unreadable unless you unencode it again. 

The only thing you need to remove from the raw certificate is "-----BEGIN CERTIFICATE-----", "-----END CERTIFICATE-----" and line breaks.

**Comment by rtribotte** [comment]
Hello @DanTheManSWE,

> The certificate is already encoded in base64 which is ok to send in a header.
If you use an octet percent-encoding on the base64 string it will replace "+" with "%2B" and "/" with "%2F" which make the certificate unreadable unless you unencode it again.

Indeed, the certificate is escaped (as mentioned in the doc), so one should definitely unescape it to obtain the initial value.

> Other reverse proxies (like Netscaler, Apache) that have the same functionality does not octet percent-encode a base64 certificate.

As far as we know, there are also other proxies that are encoding it as well (nginx).

Moreover, this has been done because characters such as `+` and `/` are delimiters/illegal in header values. If you think this is not true, can you explain why and share with us the documentation/rfc that indicates that such characters are not illegal?

**Comment by DanTheManSWE** [comment]
Hello again,
I can't find any information that `+` and `/` are illegal characters for the value.
https://www.rfc-editor.org/rfc/rfc9110.html#name-field-values

The name on the other had has more limitations.
According to Cloud Flare, this is what name and value can contain: https://developers.cloudflare.com/rules/transform/request-header-modification/reference/header-format/

**Comment by rtribotte** [comment]
Hello @DanTheManSWE,

Thanks for your feedback,

We agree on your conclusions.
We are wondering about the context, are you in a use case where unescaping the value is not possible, if yes, can you please describe it?

**Comment by DanTheManSWE** [comment]
I'm using Traefik in front of PhenixID PAS which requires a reverse proxy to do certificate authentication and then pass on the cerrtificate if you want to be able to use MTLS. It does not expect an escaped certificate.

**Comment by DanTheManSWE** [comment]
After more investigation it seems like both Apache2 and Netscaler only replace linebreaks with space. 
So my suggestion would be to add a "raw" bool option and a function to only replace linebreaks with space. 

Something like (never written anything in Go before, but I think it's correct ;) ): 
```go
func sanitize(cert []byte) string {
	cleaned := strings.NewReplacer(
		"-----BEGIN CERTIFICATE-----", "",
		"-----END CERTIFICATE-----", "",
		"\n", "",
	).Replace(string(cert))

	return url.QueryEscape(cleaned)
}

func removelinebreaks(cert []byte) string {
	cleaned := strings.NewReplacer("\n", " ").Replace(string(cert))
	return cleaned
}

func extractCertificate(ctx context.Context, cert *x509.Certificate) string {
	certPEM := pem.EncodeToMemory(&pem.Block{Type: "CERTIFICATE", Bytes: cert.Raw})
	if certPEM == nil {
		log.FromContext(ctx).Error("Cannot extract the certificate content")
		return ""
	}
	
	if raw {
		return removelinebreaks(certPEM)
	} else {
		return sanitize(certPEM)
	}
}

```

**Comment by jjtimmons** [comment]
Folks @rtribotte @kevinpollet I wanted to politely let ya'll know that this (https://github.com/traefik/traefik/pull/9412) was a breaking change for my team. We'd `url.QueryUnescape(cert)` the cert by default, which broken after URL escaping was removed in https://github.com/traefik/traefik/pull/9412

I totally get the intent of the issues + changes, I just wanted to call this out in case it may affect other users

**Comment by rtribotte** [comment]
Fixed by #9412.
